sys/riscv/riscv/pmap.c

```diff
@@ -2731,8 +2785,7 @@ pmap_enter(pmap_t pmap, vm_offset_t va, vm_page_t m, vm_prot_t prot,
 #if __has_feature(capabilities)
 	if (prot & VM_PROT_READ_CAP)
 		new_l3 |= PTE_CR;
-	if (prot & VM_PROT_WRITE_CAP)
-		new_l3 |= PTE_CW | PTE_CD;
+	new_l3 |= cheri_pte_cw(m, prot, flags);
```

Comment:
```suggestion
	if (flags & VM_PROT_WRITE_CAP)
		new_l3 |= PTE_CD;
	if (prot & VM_PROT_WRITE_CAP)
		new_l3 |= PTE_CW;
```
to mirror D/W above, no need to introduce a function for that